addons/account/tests/test_account_move_out_invoice.py

```diff
@@ -3416,3 +3416,39 @@ def test_out_invoice_reverse_caba(self):
                 'credit': value['debit'],
             })
         self.assertRecordValues(reversed_caba_move.line_ids, expected_values)
+
+    def test_tax_grid_remove_tax(self):
+        # Add a tag to tax_sale_a
+        tax_line_tag = self.env['account.account.tag'].create({
+            'name': "Tax tag",
+            'applicability': 'taxes',
+            'country_id': self.company_data['company'].country_id.id,
+        })
+
+        repartition_line = self.tax_sale_a.invoice_repartition_line_ids.filtered(
+            lambda x: x.repartition_type == 'tax')
+        repartition_line.write({'tag_ids': [(4, tax_line_tag.id, 0)]})
+
+        # Create the invoice
+        invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'invoice_date': fields.Date.today(),
+            'partner_id': self.partner_a.id,
+            'invoice_cash_rounding_id': self.cash_rounding_b.id,
+            'invoice_payment_term_id': self.pay_terms_a.id,
+            'invoice_line_ids': [
+                (0, 0, {
+                    'product_id': self.product_a.id,
+                    'price_unit': 999.99,
+                    'tax_ids': [(6, 0, self.product_a.taxes_id.ids)],
+                    'product_uom_id': self.product_a.uom_id.id,
+                }),
+            ],
+        })
+
+        with Form(invoice) as biggest_tax_invoice_form:
+            with biggest_tax_invoice_form.invoice_line_ids.edit(0) as line_form:
+                line_form.tax_ids.clear()
+
+        self.assertFalse(invoice.line_ids.mapped('tax_tag_ids'),
+                         'Tax grid on lines should be cleared when tax is removed from invoice line')
```

Comment:
```suggestion
        # Tags should be empty since the tax has been removed from the invoice line
        self.assertRecordValues(invoice.line_ids', [{'tax_tag_ids': []}])
```